src/SprykerEco/Zed/Unzer/Business/Refund/UnzerRefundExpander.php

```diff
@@ -147,27 +147,36 @@ protected function getChildUnzerCredentialsCollection(UnzerCredentialsTransfer $
     }
 
     /**
-     * @param \Generated\Shared\Transfer\PaymentUnzerTransfer $paymentUnzerTransfer
+     * @param \Generated\Shared\Transfer\RefundTransfer $refundTransfer
      * @param \Generated\Shared\Transfer\ExpenseTransfer $expenseTransfer
      *
-     * @return \Generated\Shared\Transfer\UnzerRefundTransfer
+     * @return \Generated\Shared\Transfer\RefundTransfer
      */
-    protected function createMarketplaceUnzerRefundTransfer(
-        PaymentUnzerTransfer $paymentUnzerTransfer,
+    protected function expandUnzerRefundTransfersWithExpenseTransfers(
+        RefundTransfer $refundTransfer,
         ExpenseTransfer $expenseTransfer
-    ): UnzerRefundTransfer {
-        return (new UnzerRefundTransfer())
-            ->setIsMarketplace(true)
-            ->setPaymentId($paymentUnzerTransfer->getPaymentIdOrFail())
-            ->setChargeId($expenseTransfer->getUnzerChargeIdOrFail())
-            ->addItem(
-                (new UnzerRefundItemTransfer())
-                    ->setBasketItemReferenceId(
-                        sprintf(UnzerConstants::UNZER_BASKET_SHIPMENT_REFERENCE_ID_TEMPLATE, $expenseTransfer->getUnzerParticipantIdOrFail()),
-                    )
-                    ->setQuantity(UnzerConstants::PARTIAL_REFUND_QUANTITY)
-                    ->setAmountGross($expenseTransfer->getRefundableAmountOrFail() / UnzerConstants::INT_TO_FLOAT_DIVIDER),
-            );
+    ): RefundTransfer {
+        foreach ($refundTransfer->getUnzerRefunds() as $unzerRefundTransfer) {
+            if ($unzerRefundTransfer->getChargeId() === $expenseTransfer->getUnzerChargeId()) {
+                $unzerRefundTransfer->addItem($this->createUnzerRefundItemTransfer($expenseTransfer));
+            }
+        }
+
+        return $refundTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ExpenseTransfer $expenseTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerRefundItemTransfer
+     */
+    protected function createUnzerRefundItemTransfer(ExpenseTransfer $expenseTransfer): UnzerRefundItemTransfer
```

Comment:
```suggestion
    protected function createExpenseUnzerRefundItemTransfer(ExpenseTransfer $expenseTransfer): UnzerRefundItemTransfer
```

Reply:
Doesn't it make sense to name the method in such a way that it says what is being created?

Reply:
Fixed.